src/types/bigint.rs

```diff
@@ -0,0 +1,107 @@
+use super::{private::ValueInternal, Value};
+use crate::context::internal::Env;
+use crate::context::Context;
+use crate::handle::{internal::TransparentNoCopyWrapper, Handle, Managed};
+use neon_runtime;
+use neon_runtime::raw;
+use std::fmt;
+use std::fmt::Debug;
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+#[cfg_attr(docsrs, doc(cfg(feature = "napi-6")))]
+pub struct JsBigInt(raw::Local);
+
+impl Value for JsBigInt {}
+
+unsafe impl TransparentNoCopyWrapper for JsBigInt {
+    type Inner = raw::Local;
+
+    fn into_inner(self) -> Self::Inner {
+        self.0
+    }
+}
+
+impl Managed for JsBigInt {
+    fn to_raw(&self) -> raw::Local {
+        self.0
+    }
+
+    fn from_raw(_: Env, h: raw::Local) -> Self {
+        JsBigInt(h)
+    }
+}
+
+#[derive(Debug)]
+pub struct GetBigIntLossyValueResult<T> {
+    pub value: T,
+    /// Indicates whether the BigInt value was converted losslessly
+    pub lossless: bool,
+}
+
+impl<T> fmt::Display for GetBigIntLossyValueResult<T>
+where
+    T: fmt::Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.value)
+    }
+}
+
+impl Into<i64> for GetBigIntLossyValueResult<i64> {
```

Comment:
Typically `From` is implemented instead of `Into` since there is a reflexive implementation of `Into`.

```suggestion
impl From<GetBigIntLossyValueResult<i64>> for i64 {
    fn from(other: GetBigIntLossyValueResult<i64>) -> Self {
        self.value
    }
}
```